modules/vue/tests/playwright/noHydration.spec.js

```diff
@@ -0,0 +1,111 @@
+import { test, expect } from '@playwright/test'
+import { makeProject } from '@storefront-x/testing'
+
+test('noHydration after ssr hydration', async ({ page }) => {
+  await makeProject(
+    {
+      modules: [
+        '@storefront-x/base',
+        '@storefront-x/vue',
+        '@storefront-x/vue-router',
+        [
+          'my-module',
+          {
+            components: {
+              'Hello.vue': `
+                  <template>
+                    <h1>{{ title }}</h1>
+                  </template>
+
+                  <script setup lang="ts">
+                  import { onMounted, ref } from 'vue'
+
+                  const title = ref('Hello')
+
+                  onMounted(() => {
+                    title.value = 'Hello on client'
+                  })
+                  </script>
+                `,
+            },
+            pages: {
+              'index.vue': `
+                <template>
+                  <Hello />
+                </template>
+                <script setup lang="ts">
+                import noHydrate from '#ioc/utils/hydration/noHydrate'
+                //import Hello from '#ioc/components/Hello'
+                const Hello = noHydrate(() => import('#ioc/components/Hello'))
+                </script>
+              `,
+            },
+          },
+        ],
+      ],
+    },
+    async ({ url }) => {
+      await page.goto(url, { waitUntil: 'networkidle' })
+      await expect(await page.content()).toContain('Hello')
```

Comment:
This is a bad assert because if the onMounted is executed (which is undesirable), this assert still pases because Hello on client contains Hello.

Reply:
Actually this is super bad assert because page.content() returns initial HTML returned from the server. You need to do a DOM check.